storm-client/src/storm.thrift

```diff
@@ -836,3 +848,33 @@ exception HBAuthorizationException {
 exception HBExecutionException {
   1: required string msg;
 }
+
+struct SupervisorAssignments {
+    1: optional map<string, Assignment> storm_assignment = {}
+}
+
+service Supervisor {
+  /**
+   * Send node specific assignments to supervisor
+   */
+  void sendSupervisorAssignments(1: SupervisorAssignments assignments) throws (1: AuthorizationException aze);
+  /**
+   * Get local assignment for a storm
+   */
+  Assignment getLocalAssignmentForStorm(1: string id) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  /**
+   * Send worker heartbeat to local supervisor
+   */
+  void sendSupervisorWorkerHeartbeat(1: SupervisorWorkerHeartbeat heartbeat) throws (1: AuthorizationException aze);
+}
+
+struct SupervisorWorkerHeartbeat {
```

Comment:
Looks like SupervisorWorkerHeartbeat and SupervisorWorkerHeartbeats are used "before", forward reference. 

We encountered the issue because of forward reference of thrift definition. https://issues.apache.org/jira/browse/STORM-1842

Could you rearrange the order so that all the structs are defined earlier before using them?

Reply:
You could check importing ttypes.py to see it works fine. If current patch doesn't break ttypes.py please ignore my comment.

Reply:
@HeartSaVioR 
I have tweaked the position of structs in storm.thrift.